### PR TITLE
[4.0] metismenu upgrade

### PR DESCRIPTION
**Pull request by brianteeman**
Metismenu has been updated from 1.0.3 to 1.1.0
Unfortunately they didnt follow semver so this PR is to update our code to make sure the menu still works - specifically the open and closed submenu

### CODE REVIEW

**Comment by ciar4n** [comment]
For reference the metismenu script was added here.. https://github.com/joomla/joomla-cms/pull/22587

**Comment by C-Lodder** [comment]
@brianteeman Just looked at the plugin settings and here's what I saw:

**This:**
```
var ClassNames = {
  activeClass: "active",
  collapseClass: "collapse",
  collapseInClass: "in",
  collapsingClass: "collapsing"
}
```
**was replaced with:**
```
var ClassName = {
  ACTIVE: "mm-active",
  COLLAPSE: "mm-collapse",
  COLLAPSED: "mm-collapsed",
  COLLAPSING: "mm-collapsing",
  METIS: "metismenu",
  SHOW: "mm-show"
};
```

So there are other classes that need updating.

semver at it's finest

File: build/media_source/mod_menu/js/admin-menu.es6.js [hunk 1]
```diff
@@ -144,14 +144,14 @@
...
       if (currentUrl === link.href) {
-        link.classList.add('active');
+        link.classList.add('mm-active');
         // Auto Expand Levels
         if (!link.parentNode.classList.contains('parent')) {
           const firstLevel = closest(link, '.collapse-level-1');
           const secondLevel = closest(link, '.collapse-level-2');
-          if (firstLevel) firstLevel.parentNode.classList.add('active');
+          if (firstLevel) firstLevel.parentNode.classList.add('mm-active');
           if (firstLevel) firstLevel.classList.add('in');
```
Comment (C-Lodder) [inline build/media_source/mod_menu/js/admin-menu.es6.js]: `in` >> `mm-collapsed`

File: build/media_source/mod_menu/js/admin-menu.es6.js [hunk 2]
```diff
@@ -144,14 +144,14 @@
...
-          if (secondLevel) secondLevel.parentNode.classList.add('active');
+          if (secondLevel) secondLevel.parentNode.classList.add('mm-active');
           if (secondLevel) secondLevel.classList.add('in');
```
Comment (C-Lodder) [inline build/media_source/mod_menu/js/admin-menu.es6.js]: `in` >> `mm-collapsed`

**Comment by ciar4n** [comment]
Is it not easier to set `var ClassNames` to match the current markup/js?

**Comment by C-Lodder** [comment]
@ciar4n The variable is hardcoded in the plugin and can't be overridden unless you use a global, which is horrible.

**Comment by brianteeman** [comment]
OK - all makes sense now. The docs @wilsonge showed me would only really work if we were not also using our own js and css. I think i can fix it now. Thanks

**Comment by brianteeman** [comment]
Not going to lie and say I understand the changes I made and maybe I missed something but it seems to work

**Comment by brianteeman** [comment]
updated branch to trigger drone

**Comment by brianteeman** [comment]
Conflicts resolved

**Comment by C-Lodder** [comment]
I have tested this item :white_check_mark: successfully on c5dce2ecbf3c59ee22a883b4cef0461f82bedb60<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26275">issues.joomla.org/tracker/joomla-cms/26275</a>.</sub>

**Comment by C-Lodder** [comment]
I have tested this item :white_check_mark: successfully on c5dce2ecbf3c59ee22a883b4cef0461f82bedb60<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26275">issues.joomla.org/tracker/joomla-cms/26275</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5b3eaaff9cafac5bdecf87028fab3ec386d6da72<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26275">issues.joomla.org/tracker/joomla-cms/26275</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26275">issues.joomla.org/tracker/joomla-cms/26275</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**